File: s3-inf-ppo/laboratoria/lab05/cpp/lab05.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>
#include <vector>
#include <string>

using namespace std;

class Book {
	protected:
		string title;
	
	public:			
		Book(string title) {
			this->title = title;
		}
		
		string getTitle() {
			return title;
		}
		
		virtual string getType() const = 0;
};

class Novel : public Book {

	public:	
		Novel(string title) : Book(title) {}
		
		string getType() const {
			return "powieœæ";
		}
	
};

class Library {
	public:
		vector<Novel> novels;
		
		Library* addNovel(Novel novel) {
			this->novels.push_back(novel);
			return this;
		}
		
		vector<Novel> getNovels() {
			return novels;
		}
};

int main() {
	setlocale(LC_ALL, "");
	
	Library library = Library();
	library.addNovel(Novel("Wektor pierwszy"));
	library.addNovel(Novel("Gwiazda po gwieŸdzie"));
	library.addNovel(Novel("Jednocz¹ca Moc"));
	
	for(int i = 0; i < library.novels.size(); i++) {
		Novel novel = library.getNovels().at(i);
		cout << novel.getType() << " || " << novel.getTitle() << endl;
	}
	
	return 0;
}

File: s3-inf-ppo/laboratoria/lab05/cs/lab05.cs
using System;
using System.Collections.Generic;

public class Book
{
    protected string title;

    public Book(string title)
    {
        this.title = title;
    }

    public string GetTitle()
    {
        return this.title;
    }

    public virtual string GetType()
    {
        return "ksi¹¿ka";
    }
}

public class Novel : Book
{
    public Novel(string title) : base(title) { }

    public override string GetType()
    {
        return "powieœæ";
    }
}


public class Library
{
    public List<Novel> novels = new List<Novel>();

    public Library AddNovel(Novel novel)
    {
        novels.Add(novel);
        return this;
    }

    public List<Novel> GetNovels()
    {
        return novels;
    }
}

public class Lab05
{
    public static void Main(string[] args)
    {
        Library library = new Library();
        library.AddNovel(new Novel("Wektor pierwszy"));
        library.AddNovel(new Novel("Gwiazda po gwieŸdzie"));
        library.AddNovel(new Novel("Jednocz¹ca Moc"));

        for (int i = 0; i < library.novels.Count; i++)
        {
            Novel novel = library.GetNovels()[i];
            Console.WriteLine(novel.GetType() + " || " + novel.GetTitle());
        }

        Console.ReadKey();
    }
}
